atomic_reactor/dirs.py

```diff
@@ -0,0 +1,227 @@
+"""
+Copyright (c) 2021 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import logging
+
+from pathlib import Path
+from shutil import copyfile, copytree, ignore_patterns
+from typing import Dict, Any, List, Callable, Iterable, Optional
+
+from dockerfile_parse import DockerfileParser
+
+from atomic_reactor.constants import DOCKERFILE_FILENAME, Platform
+from atomic_reactor.source import Source
+from atomic_reactor.types import ImageInspectionData
+
+logger = logging.getLogger(__name__)
+
+
+class DockerfileNotExist(Exception):
+    """Dockerfile does not exist."""
+
+
+class BuildDir(object):
+    """Representing a directory which is specific to a platform."""
+
+    def __init__(self, path: Path, platform: Platform) -> None:
```

Comment:
I can't decide how I feel about using the Platform enum vs. a plain string. On the one hand, I do think it's good to limit the valid platforms, but on the other, a plain string would usualy be nicer to work with. I'm also not sure if we should hardcode the valid platforms, since technically they are configurable in the reactor config map.

Reply:
I meant that if we use a string for the platform name, then we have infinitely many possible values, the `Platform` enum has just 4 (and that's a good thing). But I don't know if we can really enumerate all the possible platforms in the upstream code, our instance of OSBS only handles those 4 but I'm not sure if those are the only options, even our docs mention another one: https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=armhfp#image-configuration

Reply:
Platform enum is removed and the platform is passed as string now.